mycli/packages/completion_engine.py

```diff
@@ -218,16 +218,18 @@ def suggest_based_on_last_token(token, text_before_cursor, full_text, identifier
         # Check for a table alias or schema qualification
         parent = (identifier and identifier.get_parent_name()) or []
 
+        tables = extract_tables(full_text)
         if parent:
-            tables = extract_tables(full_text)
             tables = [t for t in tables if identifies(parent, *t)]
             return [{'type': 'column', 'tables': tables},
                     {'type': 'table', 'schema': parent},
                     {'type': 'view', 'schema': parent},
                     {'type': 'function', 'schema': parent}]
         else:
-            return [{'type': 'column', 'tables': extract_tables(full_text)},
+            aliases = [t[2] or t[1] for t in tables]
```

Comment:
How about `aliases = [alias or table for (schema, table, alias) in tables]`? It's longer, but it would be more readable than `t[1]` and `t[2]`.

Reply:
You're completely right, I've copied this from another line in this file. So I think this should be generalized and rewritten like you've suggested.